source/common/router/router.cc

```diff
@@ -1273,7 +1273,7 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::ResponseHeaderMapPt
         Http::CodeStats& code_stats = httpContext().codeStats();
         code_stats.chargeBasicResponseStat(cluster_->statsScope(),
                                            config_.stats_.stat_names_.retry_,
-                                           static_cast<Http::Code>(response_code));
+                                           static_cast<Http::Code>(response_code), false);
```

Comment:
Shouldn't this use whatever config is driving this option? If so, please add a failing test for this case.

Reply:
Pushed changes without a test for `onUpstreamHeaders`, but I'm uncertain on how to best test this since current test directory looks like it is intentionally set up to mock out router logic when running unit tests on `UpstreamRequest`.

That said, refactoring this change makes both all ResponseStat() cases rely on the same config, so existing failure test should verify flag works.